updater/updater.go

```diff
@@ -51,13 +52,33 @@ func CreateOrUpdate(ctx context.Context, cli client.Client, scheme *runtime.Sche
 	}
 }
 
-func CreateOrRecreate(ctx context.Context, cli client.Client, resource client.Object) func() error {
+func CreateOrRecreate(ctx context.Context, cli client.Client, scheme *runtime.Scheme, resource client.Object) func() error {
 	return func() error {
 		log.Infof("CreateOrRecreate %s", liberator_scheme.TypeName(resource))
-		err := cli.Delete(ctx, resource)
+		deleteOptions := &client.DeleteOptions{}
+		client.PropagationPolicy(metav1.DeletePropagationBackground).ApplyToDelete(deleteOptions)
+		err := cli.Delete(ctx, resource, deleteOptions)
 		if err != nil && !errors.IsNotFound(err) {
 			return err
 		}
+
+		existing, err := scheme.New(resource.GetObjectKind().GroupVersionKind())
+		if err != nil {
+			return fmt.Errorf("internal error: %w", err)
+		}
+		objectKey := client.ObjectKeyFromObject(resource)
+
+		for {
```

Comment:
Burde denne eksplisit gi opp etter en timeout? Jeg antar at den vil gi opp fordi contexten har en timeout, men det vil være lettere å feilsøke hvis vi eksplisitt setter en grense her og logger at det er derfor ting feilet.

Må bare sørge for at denne gir opp før contexten timer ut... :slightly_smiling_face:

Reply:
Hva er timeout på contexten? Sette denne til "litt mindre enn det".. evt. sette denne til "litt lenger enn det pleier å ta å slette noe" :smile: (mangler vi noe metrikker for å mene noe om disse tallene?)

Reply:
Vi har ingen overvåking (så vidt jeg vet) om hvor lang tid en ressurs i Kubernetes bruker på å blir slettet. Og ikke så interessant å overvåke tror jeg. Som regel tar det noen sekunder å slette noe. Vi har vel et krav om at en pod blir drept hardt hvis det har gått 30s etter den har fått `sigterm`. Så går an å ha timeout-en basert på det.

Ser at timeout på en rollout er 20min! Det er ganske lenge.